### ISSUE TITLE
Crash on boot (Code Signature Invalid)

### ISSUE THREAD

**Issue by Labrium** [zed-industries/community#2176]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Launch Zed 0.109.1 on macOS Catalina (10.15.7)

### Environment

N/A

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

No Zed.log was produced, but here's the crash report:
[zed_2023-10-25-173116.crash.log](https://github.com/zed-industries/community/files/13171574/zed_2023-10-25-173116.crash.log)

**Comment by ForLoveOfCats** [comment]
Hey there! 👋 Have previous versions of Zed launched on your system in the past?

**Comment by ConradIrwin** [comment]
Looking into our stats, it seems like no-one on Catalina has been able to upgrade. We made some changes to code-signing in this release, but we don't have access to a Catalina x86 machine to test on quickly.

I've reverted the riskier half of the changes to code-sign we've made to try and narrow it down.

Would you be able to try running the build from https://drive.google.com/file/d/1IdzbL9dMkELM_Ih-PcfUh32v9tzAtQBB/view? 

(It will prompt you "this file is too big to virus scan", but the downloaded binary should have a valid signature).

Although we're happy to try and help you fix this, we don't have very many users on Catalina, so we would encourage you to upgrade when you have the ability.

**Comment by roverliang** [comment]
```
Process:               zed [58678]
Path:                  /Applications/Zed.app/Contents/MacOS/zed
Identifier:            zed
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           zed [58678]
User ID:               501

Date/Time:             2023-10-26 20:28:55.950 +0800
OS Version:            macOS 11.5.1 (20G80)
Report Version:        12
Bridge OS Version:     5.5 (18P4759a)
Anonymous UUID:        5BC99198-B3D0-4E6D-9D83-F97739032A99

Sleep/Wake UUID:       EA146A1A-3091-42CF-AB0C-373BD96302A0

Time Awake Since Boot: 100000 seconds
Time Since Wake:       2700 seconds

System Integrity Protection: enabled

Crashed Thread:        0

Exception Type:        EXC_CRASH (Code Signature Invalid)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace CODESIGNING, Code 0x1

kernel messages:

VM Regions Near 0 (cr2):
--> 
    __TEXT                      10be2b000-1130d7000    [114.7M] r-x/r-x SM=COW  

Thread 0 Crashed:
0                                	0x0000000118e27000 _dyld_start + 0

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000000  rbx: 0x0000000000000000  rcx: 0x0000000000000000  rdx: 0x0000000000000000
  rdi: 0x0000000000000000  rsi: 0x0000000000000000  rbp: 0x0000000000000000  rsp: 0x00007ffee3dd4660
   r8: 0x0000000000000000   r9: 0x0000000000000000  r10: 0x0000000000000000  r11: 0x0000000000000000
  r12: 0x0000000000000000  r13: 0x0000000000000000  r14: 0x0000000000000000  r15: 0x0000000000000000
  rip: 0x0000000118e27000  rfl: 0x0000000000000200  cr2: 0x0000000000000000
  
Logical CPU:     0
Error Code:      0x00000000
Trap Number:     0

Thread 0 instruction stream not available.

Thread 0 last branch register state not available.


Binary Images:
       0x10be2b000 -        0x1130d6fff +??? (0) <113B667D-87AC-3A7E-83B3-99A9685B12F1> (null)
       0x118e26000 -        0x118ec1fff + (852.2) <57DB2053-BFD5-3683-97C6-F1DB2A1F1D09> 

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=161.6M resident=0K(0%) swapped_out_or_unallocated=161.6M(100%)
Writable regions: Total=8544K written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=8544K(100%)
 
                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
STACK GUARD                       56.0M        1 
Stack                             8192K        1 
VM_ALLOCATE                          8K        2 
__DATA                             688K        4 
__DATA_CONST                      1696K        2 
__LINKEDIT                        46.4M        4 
__TEXT                           115.3M        2 
===========                     =======  ======= 
TOTAL                            228.0M       16 

Model: MacBookPro16,2, BootROM 1554.140.20.0.0 (iBridge: 18.16.14759.0.1,0), 4 processors, Quad-Core Intel Core i5, 2 GHz, 16 GB, SMC 
Graphics: kHW_IntelIrisPlusGraphicsItem, Intel Iris Plus Graphics, spdisplays_builtin
Memory Module: BANK 0/ChannelA-DIMM0, 8 GB, LPDDR4X, 3733 MHz, Samsung, -
Memory Module: BANK 2/ChannelB-DIMM0, 8 GB, LPDDR4X, 3733 MHz, Samsung, -
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0x7BF), wl0: Jul  7 2021 02:54:21 version 9.30.444.17.32.5.70 FWID 01-6551fb93
Bluetooth: Version 8.0.5d7, 3 services, 25 devices, 1 incoming serial ports
Network Service: Wi-Fi, AirPort, en0
USB Device: USB 3.1 Bus
USB Device: USB3.0 Hub
USB Device: USB3.0 Card Reader
USB Device: USB 10/100/1000 LAN
USB Device: USB 3.1 Bus
USB Device: USB2.0 Hub
USB Device: USB 2.0 Hub
USB Device: USB Optical Mouse
USB Device: AKKO 3087
USB Device: USB Billboard Device
USB Device: Apple T2 Bus
USB Device: Touch Bar Backlight
USB Device: Touch Bar Display
USB Device: Apple Internal Keyboard / Trackpad
USB Device: Headset
USB Device: Ambient Light Sensor
USB Device: FaceTime HD Camera (Built-in)
USB Device: Apple T2 Controller
Thunderbolt Bus: MacBook Pro, Apple Inc., 85.0
Thunderbolt Bus: MacBook Pro, Apple Inc., 85.0

```

The same question.

**Comment by ForLoveOfCats** [comment]
Please try the binary linked above and let us know if it runs :)

**Comment by Labrium** [comment]
Yes, the build from the link does run fine.

**Comment by qubitter** [comment]
Same crash on my machine (MacOS Big Sur 11.6). Version linked in GDrive above works fine. Crash report is near-identical to the one pasted above.

**Comment by ConradIrwin** [comment]
Thanks for the heads up here, and sorry for breaking this for you!

I've shipped an official v0.109.3 based on the DMG here. Please let me know if you're still seeing problems.

We'll try and figure out how to get some test machines running with Catalina and Big Sur so we can replicate this. One thing that is unclear to me is that although it seems reproducibly broken on Catalina; it's only broken for a few people on Big Sur.

@qubitter or @roverliang, can you think of any settings you might have enabled that would change how apple performs its signature signing?
